src/app/pages/system/general-settings/support/support.component.ts

```diff
@@ -150,10 +150,19 @@ export class SupportComponent implements OnInit {
   }
 
   updateProductionStatus(event: MatCheckboxChange): void {
-    let request$: Observable<boolean | SetProductionStatusDialogResult> = of(true);
+    let request$: Observable<boolean | SetProductionStatusDialogResult> = of(false);
     if (event.checked) {
       request$ = request$.pipe(
-        switchMap(() => this.matDialog.open(SetProductionStatusDialogComponent).afterClosed()),
+        switchMap(() => this.matDialog.open(SetProductionStatusDialogComponent).afterClosed().pipe(
+          tap((confirmed) => {
+            if (confirmed) {
+              return true;
+            }
+            this.isProduction = false;
+            this.cdr.markForCheck();
+            return false;
```

Comment:
It seems like a leftover from the `map` usage - `return` does not matter in `tap`.